### PR TITLE
Fix SRC_FILE::DEST_FILE#archive doesn't work in TonyClient #246

### PR DESCRIPTION
**Pull request by oliverhu**
SRC_FILE::DEST_FILE#archive doesn't work in TonyClient #246

### CODE REVIEW

File: tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java [hunk 1]
```diff
@@ -34,6 +35,8 @@
   private LocalResourceType resourceType;
```
Comment (erwa) [inline tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java]: What's the difference between `path` and `localFileName`? And how do they differ from `sourceFilePath`? Can you add some Javadoc explaining the differences?

File: tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java [hunk 2]
```diff
@@ -68,8 +84,14 @@ private void parse(FileSystem fs) throws ParseException, IOException {
         throw new ParseException("Failed to parse file: " + path);
     }
 
+    FileSystem localFs = FileSystem.getLocal(new Configuration());
```
Comment (erwa) [inline tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java]: Please don't create `new Configuration()` here. Pass in a `Configuration` object to this method if needed.

Also, you can move this inside the `if` block on lines 90-92.

File: tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java [hunk 3]
```diff
@@ -42,6 +45,14 @@ public boolean isDirectory() {
...
+    return new Path(path).toUri().getScheme() == null;
+  }
+
+  public boolean isArchive() {
+    return resourceType == LocalResourceType.ARCHIVE;
+  }
+
   public Path getSourceFilePath() {
     return sourceFilePath;
   }
```
Comment (erwa) [inline tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java]: (I wasn't able to comment directly on line 66, so I'm commenting here.) Indentation is off on line 66.

File: tony-core/src/test/resources/scripts/check_archive_file_localization.py [hunk 4]
```diff
@@ -22,7 +22,9 @@
 if not os.path.isfile('./test.zip'):
     logging.error('test.zip doesn\'t exist')
     exit(-1)
```
Comment (erwa) [inline tony-core/src/test/resources/scripts/check_archive_file_localization.py]: Will this fail now that you've changed `tony-core/src/test/resources/test.zip` to `tony-core/src/test/resources/test.zip::test20.zip` in the `testTonyResourcesFlag` test? Seems like there will no longer be a `test.zip`, only a `test20.zip`.

Comment (oliverhu) [reply tony-core/src/test/resources/scripts/check_archive_file_localization.py]: yah.. that is the reason why tests are failling. Fixing

File: tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java (hunk 1 above)
Comment (oliverhu) [reply tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java]: agreed, this is confusing, will do some housekeeping for this class

Comment (erwa) [reply tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java]: Thanks. The new name `completeResourceString` and the Javadoc make it much clearer.

**Pull request merged**